### ISSUE TITLE
Overlapping Unused Symbols

### ISSUE THREAD

**Issue by TomMD** [GaloisInc/cryptol#300]
Consider:
```
% cat H.cry
module H where

type Foo = [2]
% cat G.cry
module G where

type Foo = [1]
% cat F.cry
import G
import H
```

In other words, completely unused overlapping type definitions cause errors:

```
Loading module Cryptol
Loading module G
Loading module H
Loading module Main
[error]
    Overlapping symbols defined:
    (at ./G.cry:3:6--3:9, Foo)
    (at ./H.cry:3:6--3:9, Foo)
```

**Comment by elliottt** [comment]
Looks like it's a little bit too eager when checking for overlap in the import environment. You should only get the error if you use Foo in Main.

**Comment by elliottt** [comment]
This should be fixed in a490600df51a5ba8dcbef977f99f5d02390d292f
